listenbrainz/webserver/static/js/src/brainzplayer/ProgressBar.tsx

```diff
@@ -8,54 +10,84 @@ type ProgressBarProps = {
 };
 
 // How many milliseconds to navigate to with keyboard left/right arrows
-const keyboardStepMS: number = 5000;
-const keyboardBigStepMS: number = 10000;
+const KEYBOARD_STEP_MS: number = 5000;
+const KEYBOARD_BIG_STEP_MS: number = 10000;
+
+const EVENT_KEY_ARROWLEFT: string = "ArrowLeft";
+const EVENT_KEY_ARROWRIGHT: string = "ArrowRight";
+const EVENT_TYPE_CLICK: string = "click";
+const EVENT_TYPE_MOUSEMOVE: string = "mousemove";
+const MOUSE_THROTTLE_DELAY: number = 150;
+
+const TOOLTIP_INITIAL_CONTENT: string = "0:00";
+const TOOLTIP_TOP_OFFSET: number = 102;
 
 const ProgressBar = (props: ProgressBarProps) => {
   const { durationMs, progressMs, seekToPositionMs } = props;
+  const [tipContent, setTipContent] = React.useState(TOOLTIP_INITIAL_CONTENT);
   const progressPercentage = Math.round(
     Number((progressMs * 100) / durationMs)
   );
   const hideProgressBar = isNaN(progressPercentage) || progressPercentage <= 0;
 
-  const progressClickHandler = (
-    event: React.MouseEvent<HTMLInputElement>
-  ): void => {
-    const progressBarBoundingRect = event.currentTarget.getBoundingClientRect();
-    const progressBarWidth = progressBarBoundingRect.width;
-    const musicPlayerXOffset = progressBarBoundingRect.x;
-    const absoluteClickXPos = event.clientX;
-    const relativeClickXPos = absoluteClickXPos - musicPlayerXOffset;
-    const percentPos = relativeClickXPos / progressBarWidth;
-    const positionMs = Math.round(durationMs * percentPos);
-    seekToPositionMs(positionMs);
+  // Originally by ford04 - https://stackoverflow.com/a/62017005
+  const useThrottle = (callback: any, delay: number | undefined) => {
+    const options = { leading: true, trailing: false };
+    const callbackRef = React.useRef(callback);
+    React.useEffect(() => {
+      callbackRef.current = callback;
+    });
+    return React.useCallback(
+      throttle((...args: any) => callbackRef.current(...args), delay, options),
+      [delay]
+    );
   };
 
+  const mouseEventHandler = useThrottle(
+    (event: React.MouseEvent<HTMLInputElement>): void => {
+      const progressBarBoundingRect = event.currentTarget.getBoundingClientRect();
+      const progressBarWidth = progressBarBoundingRect.width;
+      const musicPlayerXOffset = progressBarBoundingRect.x;
+      const absoluteClickXPos = event.clientX;
+      const relativeClickXPos = absoluteClickXPos - musicPlayerXOffset;
+      const percentPos = relativeClickXPos / progressBarWidth;
+      const positionMs = Math.round(durationMs * percentPos);
+      const positionTime = millisecondsToStr(positionMs);
+
+      if (event.type === EVENT_TYPE_MOUSEMOVE) {
+        setTipContent(positionTime);
+      } else if (event.type === EVENT_TYPE_CLICK) {
+        seekToPositionMs(positionMs);
+      }
+    },
+    MOUSE_THROTTLE_DELAY
+  );
+
+  React.useEffect(() => mouseEventHandler, []);
```

Comment:
This line might be left over from another time, but I believe it is not necessary and should be removed.

Reply:
Didn't notice that, thank you!